arch/x86/core/efi.c

```diff
@@ -169,3 +175,44 @@ int arch_printk_char_out(int c)
 	return efi_console_putchar(c);
 }
 #endif
+
+#ifdef CONFIG_BOOTARGS
+const char *get_bootargs(void)
+{
+	if (!efi || !efi->efi_bootargs) {
+		return NULL;
+	}
+
+	static char efi_bootargs[CONFIG_BOOTARGS_ARGS_BUFFER_SIZE] = "\0";
+
+	const char *in = NULL;
+	char *out = efi_bootargs;
+	size_t inb = efi->efi_bootargs_size;
+	size_t outb = sizeof(efi_bootargs);
+	char *efi_bootargs_virt = NULL;
+
+	k_mem_map_phys_bare((uint8_t **)&efi_bootargs_virt, (uintptr_t)efi->efi_bootargs,
+			    efi->efi_bootargs_size, 0);
+	in = efi_bootargs_virt;
+
+	/* efi uses ucs-2, we have to convert it to utf-8 */
+	iconv_t ucs2_to_u8 = iconv_open("UTF-8", "UCS-2LE");
+	size_t iconv_ret = iconv(ucs2_to_u8, &in, &inb, &out, &outb);
```

Comment:
Hm... IIRC minimal libc doesn't even have iconv, which seems like it would be a non-starter here.

Do you really need to do this, anyway?  It's a kernel command line, not an i18n-sensitive context.  IIRC Linux demands that the command line be 7-bit ASCII (or at least literally nothing assumes otherwise), can't we do the same and just implement this with a range test and cast to a char?

Reply:
removed